### PR TITLE
Feature/396 upgrade angular

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

**Comment by kcinay055679** [comment]
Upgrade done.
example dialog tests fail because of an error related to [proxy zone](https://github.com/thymikee/jest-preset-angular/issues/2162)

File: frontend/src/app/shared/dialog/example-dialog/error-messages.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (tuxmatta) [inline frontend/src/app/shared/dialog/example-dialog/error-messages.json:1]: remove this file

File: frontend/src/app/shared/dialog/example-dialog/example-dialog.component.spec.ts [hunk 2]
```diff
@@ -107,32 +137,46 @@ describe('ExampleDialogComponent', () => {
...
+    await Promise.all([nameInput, matSelect, radioButtons]).then(
+      fakeAsync(
+        ([nameInput, matSelect, radioButtons]: [MatInputHarness, MatSelectHarness, MatRadioButtonHarness[]]) => {
+          nameInput.setValue('Name');
+          advance();
+          matSelect.open().then(() => {
+            matSelect.getOptions().then((options: MatOptionHarness[]) => {
+              options[1].click();
+              radioButtons[1].check();
+              advance(100);
```
Comment (tuxmatta) [inline frontend/src/app/shared/dialog/example-dialog/example-dialog.component.spec.ts:161]: remove `100`

File: frontend/tsconfig.json [hunk 3]
```diff
@@ -16,11 +16,13 @@
     "experimentalDecorators": true,
     "moduleResolution": "node",
     "importHelpers": true,
-    "target": "es2020",
-    "module": "es2020",
-    "lib": ["es2020", "dom"],
+    "target": "ES2022",
```
Comment (tuxmatta) [inline frontend/tsconfig.json:19]: change to `es2022`

**Review by tuxmatta — CHANGES_REQUESTED** [review]

**Review by tuxmatta — APPROVED** [review]

**Pull request merged**